state/validation.go

```diff
@@ -132,12 +132,17 @@ func validateBlock(evidencePool EvidencePool, stateDB dbm.DB, state State, block
 
 	// Validate all evidence.
 	for _, ev := range block.Evidence.Evidence {
+		if evidencePool != nil {
+			if evidencePool.IsCommitted(ev) {
+				return types.NewErrEvidenceInvalid(ev, errors.New("evidence was already committed"))
+			}
+			if evidencePool.IsPending(ev) {
+				return nil
```

Comment:
not continue?

Reply:
yeah good catch, for some reason I forgot about the for loop

Reply:
this would be a signal for me that the unit test is needed 🧪

Reply:
yeah the current tests are quite basic atm i.e. just one evidence - I can further expand.

Reply:
Do you want me to consolidate the mocks into one and try use mockery?

Reply:
try to use mockery